frontend/components/search/YearSelect.jsx

```diff
@@ -0,0 +1,48 @@
+import TextField from "./StyledTextField";
+import {Grid, MenuItem} from "@material-ui/core";
+import {useContext} from "react";
+import {SearchFormContext} from "./SearchForm";
+
+export default function YearSelect({label, name}) {
+  // A component for year range inputs in the search form.
+  // `label` is displayed in the number input.
+  // `name` is the query parameter key used in the search API request.
+
+  const {state, setStateFromEvent, disabled} = useContext(SearchFormContext);
+
+  // `valueName` is the query param key for the year value (number).
+  // `typeName` is the query param key for the value type (e.g. "CE").
+  // These formats were chosen only because the API already supports them.
+  const valueName = `${name}_0`;
+  const typeName = `${name}_1`;
```

Comment:
Replacing `_0` & `_1` suffixes with something like `_value` & `_type` would produce more readable/self-explanatory param keys.

Reply:
I agree. I'll create a Jira task for that.

Reply:
78ca763c6d7357be1706eaa0c7913e9c812d3d38